__tests__/index.test.js

```diff
@@ -194,3 +194,51 @@ describe('GET /api/articles',()=>{
 });
 
 
+describe('GET /apri/articles (topic query',()=>{
+    test('return a 200 code when a request is made to the api',()=>{
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+    })
+    test('return a 200 code and allarticles filtered by query when a valid request is made to the api "mitch"',()=>{
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(12)
+        })
+    })
+    test('return a 200 code and all articles filtered by query when a valid request is made to the api "cats"',()=>{
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(1)
+        })
+    })
+    test('return a 200 code and all articles filtered by query when a valid request is made to the api "paper"',()=>{
+        return request(app)
+        .get('/api/articles?topic=paper')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(0)
+        })
+    })
+    test('return a 404 code and msg when passed a valid by non-existant query',()=>{
+        return request(app)
+        .get('/api/articles?topic=trees')
+        .expect(404)
+        .then((response)=>{
+            expect(response.body.msg).toBe('No topic with that id')
+        })
+    })
+    test('return a 200 code when passed an empty query, eg. query would be looking for an topic= empty string',()=>{
+        return request(app)
+        .get('/api/articles?topic=')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(0)
+        })
+    })
+
+})
```

Comment:
Some good considerations here, just needs some tidying up based on the above comments.
We also need to consider an additional scenario, how about if the topic in the query is valid AND does exist but there happens to be no articles associated with that valid query. What would happen then?

Reply:
Is this in reference to if topic=paper for example?  I have tested for that, returns an empty array... is this not correct?

Reply:
NOTE-discussed and agreed with Rich via slack